osgtest/tests/test_460_stashcache.py

```diff
@@ -0,0 +1,94 @@
+import os
+import pwd
+import random
+import tempfile
+
+from osgtest.library import core
+from osgtest.library import files
+from osgtest.library.osgunittest import OSGTestCase
+from osgtest.library import service
+try:
+    from urllib2 import urlopen
+except ImportError:
+    from urllib.request import urlopen
+
+
+_NAMESPACE = "stashcache"
+
+def _getcfg(key):
+    return core.config["%s.%s" % (_NAMESPACE, key)]
+
+def _setcfg(key, val):
+    core.config["%s.%s" % (_NAMESPACE, key)] = val
+
+
+class TestStashCache(OSGTestCase):
+    # testfiles with random contents
+    testfiles = [
+        ("testfile%d" % x, str(random.random()))
+        for x in range(4)
+    ]
+
+    def assertCached(self, name, contents):
+        fpath = os.path.join(_getcfg("cache_dir"), name)
+        self.assertTrue(os.path.exists(fpath),
+                        name + " not cached")
+        self.assertEqual(contents, files.read(fpath, as_single_string=True),
+                         "cached file %s contents do not match expected" % name)
```

Comment:
Should we include the actual content vs the expected content in the message?

Reply:
Sure. I added a new `assertEqualVerbose` method that prints the expected and actual values on failure.